From 6a13874140a69909daa59d826c73b382c60e8169 Mon Sep 17 00:00:00 2001
From: Zhou Guanghui <zhouguanghui1@huawei.com>
Date: Mon, 19 Sep 2022 07:32:14 +0000
Subject: [PATCH 061/595] [Huawei] mm/sharepool: fix the incorrect judgement of
 the addr range

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 187892

----------------------------------------------

The address range of dvpp is [start, start + size), the value of
start + size can be out of the address range.

Signed-off-by: Zhou Guanghui <zhouguanghui1@huawei.com>
---
 mm/share_pool.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/share_pool.c b/mm/share_pool.c
index d079c5cbbfe9..8e51ca5327b1 100644
--- a/mm/share_pool.c
+++ b/mm/share_pool.c
@@ -3635,7 +3635,7 @@ bool mg_sp_config_dvpp_range(size_t start, size_t size, int device_id, int pid)
 	/* NOTE: check the start address */
 	if (pid < 0 || size <= 0 || size > MMAP_SHARE_POOL_16G_SIZE ||
 	    device_id < 0 || device_id >= MAX_DEVID || !is_online_node_id(device_id)
-		|| !is_sp_dynamic_dvpp_addr(start) || !is_sp_dynamic_dvpp_addr(start + size))
+		|| !is_sp_dynamic_dvpp_addr(start) || !is_sp_dynamic_dvpp_addr(start + size - 1))
 		return false;
 
 	ret = get_task(pid, &tsk);
-- 
2.17.1

